git-config/src/file/git_config.rs

```diff
@@ -1673,6 +1770,190 @@ mod from_paths {
+        let non_canonical_path_a = format!("{}/./a", a_path_string);
+        let non_existing_path = "/dfgwfsghfdsfs";
+        let c_path = dir.path().join("c");
+        fs::write(
+            c_path.as_path(),
+            format!(
+                "
+            [core]
+              c = 12
+              d = 42
+            [include]
+              path = {}
+              path = {}
+              path = {}
+            [http]
+	          sslVerify = false",
+                non_existing_path,
+                non_canonical_path_a,
+                relative_b_path.as_path().to_str().unwrap()
+            ),
+        )
+        .unwrap();
+
+        let config = GitConfig::from_paths(vec![c_path], &Default::default()).unwrap();
+
+        assert_eq!(
+            config.get_raw_value("core", None, "c"),
+            Ok(Cow::<[u8]>::Borrowed(b"12"))
+        );
+        assert_eq!(
+            config.get_raw_value("core", None, "d"),
+            Ok(Cow::<[u8]>::Borrowed(b"41"))
+        );
+        assert_eq!(
+            config.get_raw_value("http", None, "sslVerify"),
+            Ok(Cow::<[u8]>::Borrowed(b"false"))
+        );
+
+        assert_eq!(
+            config.get_raw_value("diff", None, "renames"),
+            Ok(Cow::<[u8]>::Borrowed(b"true"))
+        );
+
+        assert_eq!(
+            config.get_raw_value("core", None, "a"),
+            Ok(Cow::<[u8]>::Borrowed(b"false"))
+        );
+    }
+
+    #[test]
+    fn nested_include_respects_max_depth() {
+        let dir = tempdir().unwrap();
+        let max_depth = 3;
+        let options = from_paths::Options {
+            git_install_dir: None,
+            max_depth,
+        };
+        let path = dir.path().join("0");
+        fs::write(
+            path.as_path(),
+            format!(
+                "
+                [core]
+                  i = true 
+                [include]
+                  path = {}",
+                to_backslash_escaped_string(&path)
+            ),
+        )
+        .unwrap();
+
+        for (i, prev_i) in (1..=max_depth).zip(0..max_depth) {
+            let path = dir.path().join(i.to_string());
+            let prev_path = dir.path().join(prev_i.to_string());
+            fs::write(
+                path.as_path(),
+                format!(
+                    "
+                    [core]
+                      i = false 
+                    [include]
+                      path = {}",
+                    to_backslash_escaped_string(&prev_path)
+                ),
+            )
+            .unwrap();
+        }
+
+        let config = GitConfig::from_paths(vec![dir.path().join((max_depth - 1).to_string())], &options).unwrap();
+        assert_eq!(
+            config.get_raw_value("core", None, "i"),
+            Ok(Cow::<[u8]>::Borrowed(b"true"))
+        );
+
+        let config = GitConfig::from_paths(vec![dir.path().join(max_depth.to_string())], &options).unwrap();
```

Comment:
I know you tested it, but when looking at [the code in git](https://github.com/git/git/blob/master/config.c#L183:L183) I noticed that it can indeed abort if the max-depth is exceeded.

Hence I think there should be an error instead of a potentially different value. This seems fair now especially since `max_depth` can be configured.

Reply:
Great you saw it, I never tested this. What I manually tested with git was including non-existing path which is ignored and we handle it. 

I think we had discussion to not have max depth at all, then we decided to add it and I didn't really look at how git is handling it as I should have. 

I'll add MaxDepth quick error and return that instead.

Reply:
Awesome, thanks!

Good it was caught, and thanks for the refresher on non-existing files, already forgot about it.